transforms/helpers/transform-helper.ts

```diff
@@ -161,25 +161,28 @@ function createMethodProp(
     decorators = [],
   }: { isAction?: boolean; decorators?: Decorator[] } = {}
 ): ClassMethod {
-  const propKind =
+  const kind =
     functionProp.kind === 'init' ? 'method' : defined(functionProp.kind);
 
-  return withDecorators(
-    withComments(
-      replaceSuperExpressions(
-        j,
-        j.classMethod(
-          propKind,
-          functionProp.key,
-          functionProp.params,
-          functionProp.body
-        ),
-        functionProp,
-        { isAction }
-      ),
-      functionProp
-    ),
-    decorators
+  const existingDecorators =
+    functionProp instanceof EOMethodProp
```

Comment:
Any way to make this polymorphic? Something seems off about this to me.

Reply:
I'm not sure. I tried 10 different ways to get rid of `FunctionProp`, which is a POJO masquerading as an `EOMethodProp`, but it's just different enough re: mutability that I eventually gave up. It really threw the initial TS conversion off track too because it wasn't clear for a long while that the POJO thing was happening. Super confusing.